### DISCUSSION TITLE
Deploy without node_modules folder??

### DISCUSSION THREAD

**Discussion by zoran123456** [vercel/next.js#20524] [Help]
Hi, I am deploying an app to Azure with source code. However "node_modules" is a huge folder and its majority is completely useless for my app.

I tried to deploy without this folder, however, the application failed on Azure. So is there a way to deploy without this folder, or just to extract the bare minimum for the "next js" app to work? If so, what are the required node modules for the next-js app?

**Comment by GuillaumeSarfati** [comment]
Usually you exclude the node_modules folder and you use the command `npm install` on your server to install the dependencies
https://docs.microsoft.com/en-us/azure/app-service/quickstart-nodejs?pivots=platform-linux

**Reply by RayanMM** [reply]
The problem is when it's running the release, it takes a lot of time to finish copying the files to the server. When commiting the node_modules folder is on .gitignore file. So when a new merge is made the pipeline runs with all necessary commands (npm install, npm run build...) and then the release pipeline is started. At this point it takes sometimes about 1 hour to conclude

**Comment by RayanMM** [comment]
I'm going through the same issue. The deploy take a long time to finish because of node_modules folder, as the app gets more features the time to deploy increases as well. I hope to find a way to solve this.

**Comment by ayush2708** [comment]
Hi, any update on this. Facing same issue. The thing is I am using Azure Pipelines to deploy the App onto App service via Agent. The task doc says only zip package upto 1GB is supported. We have reached a size upto 800mb and I am concerned that I would not be able to deploy if it's exceeds 1GB. So need some input. Thank you

**Comment by pamionj** [comment]
This whole node_modules BS suck so much. My nextjs web app was like 900 MB with them and only 41 mb after deleting node_modules but I have no idea if it will work when I host it on Vercel/Whatever.
